README.md

```diff
@@ -266,6 +266,32 @@ extern "Rust" {
     /// `ptr` has to point to the beginning of an allocated block.
     fn miri_static_root(ptr: *const u8);
 
+    /// Miri-provided extern function to obtain a backtrace of the current call stack.
+    /// This returns a boxed slice of pointers - each pointer is an opaque value
+    /// that is only useful when passed to `miri_resolve_frame`
+    fn miri_get_backtrace() -> Box<[*mut ()]>;
+
+    /// Miri-provided extern function to resolve a frame pointer obtained
+    /// from `miri_get_backtrace`. The `version` argument must be `0`,
+    /// and `MiriFrame` should be declared as follows:
+    ///
+    /// ```rust
+    /// struct MiriFrame {
```

Comment:
Shouldn't this be `repr(C)`?

Reply:
We use `mplace_field` to write to fields, which ends up using `FieldsShape` to map the definition-order fields to their actual offsets in memory.

Reply:
Oh so we exploit that we actually have access to the user-defined type... sneaky.^^